servers/parser-graphql-wrapper/src/preview/preview.integration.ts

```diff
@@ -0,0 +1,198 @@
+import nock, { cleanAll } from 'nock';
+import { getRedis } from '../cache';
+import { startServer } from '../server';
+import { ApolloServer } from '@apollo/server';
+import request from 'supertest';
+import { print } from 'graphql';
+import { gql } from 'graphql-tag';
+import { IContext } from '../context';
+import { Application } from 'express';
+import { DataSource } from 'typeorm';
+import { getConnection, getSharedUrlsConnection } from '../database/mysql';
+import { ItemResolver } from '../entities/ItemResolver';
+import { IntMask } from '@pocket-tools/int-mask';
+import * as ogs from 'open-graph-scraper';
+import { mockUnleash } from '@pocket-tools/feature-flags-client';
+import * as unleash from '../unleash';
+import config from '../config';
+
+jest.mock('open-graph-scraper');
+
+describe('preview', () => {
+  let app: Application;
+  let server: ApolloServer<IContext>;
+  let graphQLUrl: string;
+  let connection: DataSource;
+  const { unleash: mockClient, repo } = mockUnleash([]);
+
+  const GET_PREVIEW = gql`
+    query display($url: String!) {
+      itemByUrl(url: $url) {
+        preview {
+          id
+          title
+          image {
+            url
+            src
+          }
+          excerpt
+          authors {
+            id
+          }
+          domain {
+            name
+            logo
+          }
+          datePublished
+          url
+          item {
+            id
+          }
+        }
+      }
+    }
+  `;
+
+  const testUrl = 'https://test.com';
+
+  const item = {
+    itemId: 123,
+    searchHash: '123455sdf',
+    normalUrl: testUrl,
+    resolvedId: 123,
+    hasOldDupes: false,
+  };
+
+  const parserItemId = '123';
+
+  const defaultExpected = {
+    id: 'encodedId',
+    image: null,
+    excerpt: null,
+    authors: null,
+    domain: { logo: null, name: 'test.com' },
+    datePublished: null,
+    url: testUrl,
+    item: {
+      id: 'encodedId',
+    },
+  };
+
+  const openGraphFeatureToggle = {
+    name: config.unleash.flags.openGraphParser.name,
+    stale: false,
+    type: 'release',
+    project: 'default',
+    variants: [],
+    strategies: [],
+    impressionData: false,
+  };
+
+  beforeAll(async () => {
+    jest.spyOn(unleash, 'unleash').mockReturnValue(mockClient);
+    ({ app, server, url: graphQLUrl } = await startServer(0));
+    connection = await getConnection();
+    //Delete the items
+    const entities = connection.entityMetadatas;
```

Comment:
nit, but can this be a truncate statement or something?

Reply:
done!